autoload/go/cmd.vim

```diff
@@ -183,14 +184,19 @@ function! go#cmd#Test(bang, compile, ...)
         " expand all wildcards(i.e: '%' to the current file name)
         let goargs = map(copy(a:000), "expand(v:val)")
 
-        " escape all shell arguments before we pass it to test
-        call extend(args, go#util#Shelllist(goargs, 1))
+        call extend(args, goargs, 1)
```

Comment:
I guess that a recent change 579dd89 broke go#cmd#GoTestFunc because go#util#Shelllist also escapes "-run" flag.
GoTestFunc passes `goargs` that contains "-run=TestABC" but `go#util#Shelllist` escapes that flag.